### DISCUSSION TITLE
Does libp2p work on Android?

### DISCUSSION THREAD

**Discussion by RemonAhammad** [libp2p/rust-libp2p#5669] [Q&A] [answered]
Does libp2p work on Android?
Basically, I want to run libp2p using FFI for Android phones. Is it suitable for Android or iOS phones?

**Comment by dariusc93** [comment, marked as answer]
Hey! rust-libp2p does work on android and ios (although it been awhile since ive tested it on there so stuff may have changed since then) however you would need to write your own ffi bindings, etc.

**Reply by RemonAhammad** [reply]
worked.

**Reply by ijsnow** [reply]
Could you provide or point me at an example? I’m getting some mysterious linker issues when building for ios

https://github.com/libp2p/rust-libp2p/issues/5679

**Reply by RemonAhammad** [reply]
try with `cargokit`
